src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -598,6 +599,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._userNameInputBox.enable();
 			this._passwordInputBox.enable();
 			this._rememberPasswordCheckBox.enabled = true;
+
+			this._initialConnectionInfo.authenticationType = AuthenticationType.SqlLogin;
+			if (this._initialConnectionInfo && this._initialConnectionInfo.userName) {
+				const loadProfileIntoPasswordInputBox = (profile: IConnectionProfile) => {
+					this._passwordInputBox.value = profile.password;
+				};
+
+				this._rememberPasswordCheckBox.checked = this._initialConnectionInfo.savePassword;
+				this._connectionManagementService.addSavedPassword(this._initialConnectionInfo, true).then(loadProfileIntoPasswordInputBox);
```

Comment:
let's use `await` instead.

Reply:
This seems like it's more trouble than it's worth since the call chain isn't small and not awaiting something correctly could cause issues, IMO.